hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/TestOmMetadataManager.java

```diff
@@ -689,12 +690,17 @@ public void testListSnapshot() throws Exception {
     }
 
     //Test listing snapshots with no volume name.
-    Assert.assertThrows(OMException.class, () -> omMetadataManager.listSnapshot(
-        null, null));
+    Assert.assertThrows(OMException.class,
```

Comment:
It would be great if you separate out success and failure scenarios.  Also use [parameterized test](https://www.baeldung.com/parameterized-tests-junit-5).

Reply:
In junit4, if we want to write a parameterized test in `TestOmMetadataManager`, we must use `@RunWith(Enclosed.class)`.
It might change the whole class. I'm not sure if it is necessary or there is other way to do it.

Reply:
Suggestion is to use Junit-5. We already have [Junit-5 dependency in Ozone-manager](https://github.com/apache/ozone/blob/HDDS-6517-Snapshot/hadoop-ozone/ozone-manager/pom.xml#L211-L215). You don't need @RunWith in Junit-5. You can use [TestOzoneManagerDoubleBuffer](https://github.com/apache/ozone/blob/HDDS-6517-Snapshot/hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/ratis/TestOzoneManagerDoubleBuffer.java#L180) for reference. 

You have to do following changes along with test:
1. Change [@Before to @BeforeEach](https://github.com/apache/ozone/blob/HDDS-6517-Snapshot/hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/ratis/TestOzoneManagerDoubleBuffer.java#L77).
2. Change [@After to @AfterEach](https://github.com/apache/ozone/blob/HDDS-6517-Snapshot/hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/ratis/TestOzoneManagerDoubleBuffer.java#L128).
3. Change [@Rule to @TempDir](https://github.com/apache/ozone/blob/HDDS-6517-Snapshot/hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/ratis/TestOzoneManagerDoubleBuffer.java#L74).

Rest of the tests will work as it is.

Reply:
Done, thanks.